frontend/src/app/app.tsx

```diff
@@ -1,13 +1,18 @@
 import { actions as authActions } from '~/bundles/auth/store/auth.js';
 import { RouterOutlet } from '~/bundles/common/components/components.js';
 import { useAppDispatch, useEffect } from '~/bundles/common/hooks/hooks.js';
+import { storage, StorageKey } from '~/framework/storage/storage.js';
 
 const App: React.FC = () => {
     const dispatch = useAppDispatch();
 
     useEffect(() => {
-        void dispatch(authActions.loadCurrentUser());
-    });
+        void storage.has(StorageKey.TOKEN).then((hasToken) => {
+            if (hasToken) {
+                void dispatch(authActions.loadCurrentUser());
+            }
+        });
+    }, [dispatch]);
```

Comment:
When there is no token is user redirected to sign-in? I am asking because in [update protected route pr](https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/pull/119) Miguel did the opposite changes to what you do here.

Reply:
Yes, because if no token, we are making unnecessary request and getting an error notification message, also in console. This way we fix it.

Reply:
We are using async-await approach